theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -227,7 +228,7 @@ def run_gradinput(self, inputs_shape, filters_shape, output_shape,
                     border_mode=border_mode, subsample=subsample,
                     conv_mode=conv_mode)
         f = theano.function([], c, mode=mode)
-        self.assertTrue(hasattr(f.maker.fgraph.outputs[0].tag, 'trace'))
+        self.assertTrue(check_stack_trace(f, ops_to_check='all'))
         f_ref = theano.function([], c_ref, mode='FAST_RUN')
 
         if target_op is not None:
```

Comment:
Have you tried running the GPU tests locally? I think some of them inherit from this class, and the check would fail for them. I think they are currently passing because the output node would be HostFromGpu, not the optimized conv op.
I think it would be better to add a "check_trace" argument to `run_fwd`, `run_gradweight`, and `run_gradinput`. You can set it to `False` by default, but to `True` in `TestCorrConv2d` and `TestCpuConv2d` below.